pennylane/qaoa/cost.py

```diff
@@ -244,3 +244,214 @@ def maxcut(graph):
     )
     H = edge_driver(graph, ["10", "01"]) + identity_h
     return (H, qaoa.x_mixer(graph.nodes))
+
+
+def max_independent_set(graph, constrained=True):
+    r"""Returns the QAOA cost Hamiltonian and the recommended mixer corresponding to the MaxIndependentSet problem,
+    for a given graph.
+
+    Given some graph :math:`G`, an independent set is a set of vertices such that no two of the vertices in the set
+    share a common edge. The goal of MaxIndependentSet is to find the largest such set.
+
+    Args:
+        graph (nx.Graph): a graph defining the pairs of wires on which each term of the Hamiltonian acts
+        constrained (bool): specifies the variant of QAOA that is performed (constrained or unconstrained)
+
+    Returns:
+        (.Hamiltonian, .Hamiltonian):
+
+    .. UsageDetails::
+
+        There are two variations of QAOA for this problem, constrained and unconstrained:
+
+        **Constrained**
+
+        .. note::
+
+            This method of constrained QAOA was introduced by Hadfield, Wang, Gorman, Rieffel, Venturelli, and Biswas
+            in `[arXiv:1709.03489] <https://arxiv.org/abs/1709.03489>`__.
+
+        The MaxIndependentSet cost Hamiltonian for constrained QAOA is defined as:
+
+        .. math:: H_C \ = \ \displaystyle\sum_{v \in V(G)} Z_{v}
+
+        where :math:`V(G)` is the set of vertices of the input graph, and :math:`Z_i` is the Pauli-Z
+        operator applied to the :math:`i`-th vertex.
+
+        The returned mixer Hamiltonian is `~qaoa.bit_flip_mixer` applied to :math:`G`.
+
+        .. note::
+
+            **Recommended initialization circuit:**
+                Each wire in the :math:`|0\rangle` state
+
+        **Unconstrained**
+
+        The MaxIndependentSet cost Hamiltonian for unconstrained QAOA is defined as:
+
+        .. math:: H_C \ = \ \frac{(i, j) \in E(G)} (Z_i Z_j \ - \ Z_i \ - \ Z_j) \ + \
+                  \displaystyle\sum_{i \in V(G)} Z_i
```

Comment:
![image](https://user-images.githubusercontent.com/49409390/93518068-516f9100-f8fa-11ea-9dcb-ef8e889bd5d4.png)
It's rendering a bit weirdly, I'm guessing the first part should be a sum?